From 8da88fa0aa9b8be9bdb67992be06b3bbcbda56f1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Kelemen=20=28vudiq=29?= <vudiq@vudiq.sk>
Date: Sun, 29 Jan 2023 10:10:24 +0100
Subject: [PATCH 1/4] cmake: remove incorrect / duplicate directories include

this is referencing `dirksimple` instead of `dirksimple_libretro`, but
same directories are already included some lines above
---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 485bbaa..58685d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,10 +197,6 @@ if(DIRKSIMPLE_LIBRETRO)
         thirdparty/theoraplay
     )
     set_target_properties(dirksimple_libretro PROPERTIES PREFIX "")
-    target_include_directories(dirksimple PRIVATE
-        thirdparty/lua/src
-        thirdparty/theoraplay
-    )
     target_link_libraries(dirksimple_libretro dirksimple-theoraplay dirksimple-theora dirksimple-vorbis dirksimple-ogg dirksimple-lua)
     
     if(USE_PTHREAD)

From 73bc34aace372a86450791e3e82ac86b3e7767c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Kelemen=20=28vudiq=29?= <vudiq@vudiq.sk>
Date: Sun, 29 Jan 2023 10:16:09 +0100
Subject: [PATCH 2/4] cmake: libretro - fix target name for pthread linking

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 58685d7..ed87f1d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,7 +200,7 @@ if(DIRKSIMPLE_LIBRETRO)
     target_link_libraries(dirksimple_libretro dirksimple-theoraplay dirksimple-theora dirksimple-vorbis dirksimple-ogg dirksimple-lua)
     
     if(USE_PTHREAD)
-        target_link_libraries(dirksimple pthread)
+        target_link_libraries(dirksimple_libretro pthread)
     endif()
 
     if(HAVE_LIB_M)

From d949d7d66bf67b10045a526af62fc38601bac6d4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Kelemen=20=28vudiq=29?= <vudiq@vudiq.sk>
Date: Sun, 29 Jan 2023 10:17:34 +0100
Subject: [PATCH 3/4] cmake: shift games and apple commands to SDL build

when building only libretro core (-DLIBRETRO=ON), this part of the
script fails, as there is no `dirksimple` target.
---
 CMakeLists.txt | 38 +++++++++++++++++++-------------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed87f1d..ad8c5b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,6 +188,25 @@ if(DIRKSIMPLE_SDL)
     if(NOT DIRKSIMPLE_FORCE_BASE_DIR STREQUAL "")
         set_target_properties(dirksimple PROPERTIES COMPILE_DEFINITIONS "DIRKSIMPLE_FORCE_BASE_DIR=${DIRKSIMPLE_FORCE_BASE_DIR}")
     endif()
+
+   add_custom_command(
+       TARGET dirksimple POST_BUILD
+       COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/games"
+   )
+   
+   if(APPLE)
+       add_custom_command(
+           TARGET dirksimple POST_BUILD
+           COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app"
+           COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents"
+           COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/MacOS"
+           COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources"
+           COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources/games"
+           COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_BINARY_DIR}/dirksimple" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/MacOS/dirksimple"
+           COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_SOURCE_DIR}/mac-appbundle-info.plist" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Info.plist"
+           COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_SOURCE_DIR}/icon.icns" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources/DirkSimple.icns"
+       )
+   endif()
 endif()
 
 if(DIRKSIMPLE_LIBRETRO)
@@ -214,24 +233,5 @@ if(DIRKSIMPLE_LIBRETRO)
     endif()
 endif()
 
-add_custom_command(
-    TARGET dirksimple POST_BUILD
-    COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/games"
-)
-
-if(APPLE)
-    add_custom_command(
-        TARGET dirksimple POST_BUILD
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app"
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents"
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/MacOS"
-        COMMAND ${CMAKE_COMMAND} ARGS -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources/games"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_BINARY_DIR}/dirksimple" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/MacOS/dirksimple"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_SOURCE_DIR}/mac-appbundle-info.plist" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Info.plist"
-        COMMAND ${CMAKE_COMMAND} ARGS -E copy "${CMAKE_CURRENT_SOURCE_DIR}/icon.icns" "${CMAKE_CURRENT_BINARY_DIR}/DirkSimple.app/Contents/Resources/DirkSimple.icns"
-    )
-endif()
-
 # end of CMakeLists.txt ...
 

From 1d3878d09415e2212a41e7dd4d848a30bf28e1dc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Kelemen=20=28vudiq=29?= <vudiq@vudiq.sk>
Date: Sun, 29 Jan 2023 10:27:06 +0100
Subject: [PATCH 4/4] cmake: cleanups

- removed code referencing mojozork
- removed whitespace
---
 CMakeLists.txt | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad8c5b7..b4e8a2e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,15 +11,10 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Android")
     set(ANDROID TRUE)
 endif()
 
-# Building as part of RetroArch? Turn off everything but the libretro plugin by default.
-if(LIBRETRO)
-    set(MOJOZORK_STANDALONE_DEFAULT OFF)
-endif()
-
 include(CheckLibraryExists)
 
 CHECK_LIBRARY_EXISTS(m floor "" HAVE_LIB_M)
-                                                                                                                                         
+
 set(DIRKSIMPLE_SDL_DEFAULT ON)
 set(DIRKSIMPLE_LIBRETRO_DEFAULT ON)
 
@@ -193,7 +188,7 @@ if(DIRKSIMPLE_SDL)
        TARGET dirksimple POST_BUILD
        COMMAND ${CMAKE_COMMAND} ARGS -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/games" "${CMAKE_CURRENT_BINARY_DIR}/games"
    )
-   
+
    if(APPLE)
        add_custom_command(
            TARGET dirksimple POST_BUILD
@@ -217,7 +212,7 @@ if(DIRKSIMPLE_LIBRETRO)
     )
     set_target_properties(dirksimple_libretro PROPERTIES PREFIX "")
     target_link_libraries(dirksimple_libretro dirksimple-theoraplay dirksimple-theora dirksimple-vorbis dirksimple-ogg dirksimple-lua)
-    
+
     if(USE_PTHREAD)
         target_link_libraries(dirksimple_libretro pthread)
     endif()
